x-pack/filebeat/input/awss3/config.go

```diff
@@ -50,7 +51,8 @@ func defaultConfig() config {
 
 func (c *config) Validate() error {
 	if c.QueueURL == "" && c.BucketARN == "" {
-		return fmt.Errorf("queue_url or bucket_arn must provided")
+		logp.NewLogger(inputName).Warnf("neither queue_url nor bucket_arn were provided, input %s will stop", inputName)
+		return nil
```

Comment:
That's because `Validate` will pass if neither `queue_url` nor `bucket_arn` is set.
I change this behaviour after report on https://github.com/elastic/beats/issues/23226#issuecomment-905255038

Since we cannot control all users' config we would risk a lot of setup updated to `7.15` starting to fail to start Filebeat because they have `aws` module defined with no all the filesets not actually used set to `enable: false`
This shouldn't affect filesets in `aws` module that are already setup properly with `queue_url`

Reply:
Yep and that's why we have `enabled: false` set in the aws.yml config for all filesets. My thought is to still keep the original behavior, stop Filebeat if the configuration is wrong. Otherwise users would start asking what are these warning logs for and the same thing, `enabled: false` is required. Also I believe the behavior is what we have for other filebeat modules too.

Reply:
@exekias WDYT? Also https://github.com/elastic/beats/pull/27526 should help with this issue too.

Reply:
@kaiyan-sheng the original behaviour is to no stop Filebeat if an aws fileset is not defined, and we already have "strange" log entries when this happens:
```
2021-08-25T12:03:34.312+0200    ERROR   [input.aws-s3]  awss3/input.go:93       getRegionFromQueueURL failed: QueueURL is not in format: https://sqs.{REGION_ENDPOINT}.{ENDPOINT}/{ACCOUNT_NUMBER}/{QUEUE_NAME} {"queue_url": "<no value>"}
```

the behaviour will be the same and the log entry will be replace with:
```
2021-08-26T16:29:12.960+0200    WARN    [aws-s3]        awss3/config.go:55      neither queue_url or bucket_arn were provided, input aws-s3 will stop
```

Reply:
The problem, as far as I can see it, is that we cannot control what aws module's config users have defined, with or without every filesets defined with `enabled: false`
The source of the problem resides on https://github.com/elastic/beats/blob/master/filebeat/fileset/modules.go#L75 and https://github.com/elastic/beats/blob/master/filebeat/fileset/modules.go#L83, where we set an empty `FilesetConfig` having `FilesetConfig.Enabled = nil` and then we check for `FilesetConfig.Enabled != nil && !(*fcfg.Enabled)`.
Changing to `fcfg = &FilesetConfig{Enabled: new(bool) }`, would be probably the right solution, but I can see a similar change in behaviour (https://github.com/elastic/beats/pull/27526/files) was decided to be breaking and will happen only on `8.x`

Excluding the change there, and assuming we don't want to have updates to `7.15` for filebeat stop starting (given custom users' config with no fileset defined), I cannot see any other way to prevent than the "fix" in this PR (I've tried changing the config rendering so to not fail to start, and even if possible in the end the result wouldn't be different than logging a strange message that something is not properly configured)

Before `queue_url` was always rendered in the fileset config, as `<no value>` if the fileset was not defined at all.
Now either `queue_url` or `bucket_arn` will be rendered according to which one of the related `var` is defined.
Since both of them can not being defined, and the fileset won't be skipped for validation, the less disruptive way I found would be to not fail config validation in this case.

The case was reported in #23226

@exekias hope you have more context

Reply:
Sorry for the late reply,

Ideally a wrong config would end up in an error, which would make Filebeat stop.

I understand the concern around the change to current behavior, and it seems we cannot implement this error in a backwards compatible way, at least for 7.15. What about doing this now and enabling the error once https://github.com/elastic/beats/pull/27526 goes in?

Reply:
>Ideally a wrong config would end up in an error, which would make Filebeat stop.

I agree, the problem here is that it could be both a true positive error (fileset defined with no queue_url or bucket_arn) or a false positive one (fileset not defined), we have to decide what's most destructive

>What about doing this change as it is now and enabling the error once #27526 goes in?

for me it's fine, but this mean targeting 8.x
@kaiyan-sheng ?

Reply:
@exekias Thanks for your input. @aspacca Targeting 8.x sounds good to me. Maybe we can add a warning in the documentation for now?